pkgs/tools/misc/diskonaut/default.nix

```
@@ -0,0 +1,39 @@
+{ stdenv, rustPlatform, fetchFromGitHub, fetchpatch }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "diskonaut";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "imsnif";
+    repo = "diskonaut";
+    rev = version;
+    sha256 = "0vnmch2cac0j9b44vlcpqnayqhfdfdwvfa01bn7lwcyrcln5cd0z";
+  };
+
+  patches = [
+    (fetchpatch {
+      name = "turn-off-parallelism-when-in-test-mode";
+      url = https://patch-diff.githubusercontent.com/raw/imsnif/diskonaut/pull/43.patch;
+      sha256 = "1gpsh3lkjh50pbb5yb2j1hl4hqr3q2g7dc6vjn7w7ic700nrlakv";
+    })
+  ];
+
+  # some tests fail due to non-portable (in terms of filesystems) measurements of block sizes
+  # see https://github.com/imsnif/diskonaut/issues/50 for more info
+  checkPhase = ''
+    cargo test -- \
+    --skip=tests::cases::ui::small_files_with_x_as_zero \
+    --skip=tests::cases::ui::small_files_with_y_as_zero
+  '';
```

Just disable tests altogether until the next release. aarch64 also still fails tests.

```suggestion
  doCheck = false;
```